### PR TITLE
Use getDriverName to identify driver

### PR DESCRIPTION
**Pull request by bebef1987**
for AppiumWebDriver:use AppiumFieldDecorator

Attempt to Fix: #872

### CODE REVIEW

File: serenity-core/src/main/java/net/thucydides/core/annotations/locators/SmartElementProxyCreator.java [hunk 1]
```diff
@@ -19,8 +19,8 @@
 	@Override
 	public void proxyElements(PageObject pageObject, WebDriver driver) {
 		if(driver instanceof WebDriverFacade) {
-			WebDriver proxiedDriver = ((WebDriverFacade) driver).getProxiedDriver();
-			if(proxiedDriver instanceof AppiumDriver) {
+			String driverName = ((WebDriverFacade) driver).getDriverName();
```
Comment (bebef1987) [inline serenity-core/src/main/java/net/thucydides/core/annotations/locators/SmartElementProxyCreator.java]: Some other solution:
```
			String driverClass = ((WebDriverFacade) driver).getDriverClass().getCanonicalName();
			if(driverClass.contains("io.appium.java_client.AppiumDriver") ) {
```

**Comment by bebef1987** [comment]
@wakaleo Would this fix our issue?

**Comment by wakaleo** [comment]
I just pushed another change that should do the trick.

**Comment by bebef1987** [comment]
@wakaleo  tested this and it does the trick. Thanks for fixing this

**Comment by wakaleo** [comment]
Great to here, thanks for the update!

**Pull request closed without merging**